tests/pytorch/test_datasets_pt.py

```diff
@@ -18,26 +18,28 @@ def test_visiondataset():
 
 
 @pytest.mark.parametrize(
-    "dataset_name, train, input_size, size, rotate",
+    "dataset_name, train, input_size, size, rotate, download",
     [
-        ['FUNSD', True, [512, 512], 149, False],
-        ['FUNSD', False, [512, 512], 50, True],
-        ['SROIE', True, [512, 512], 626, False],
-        ['SROIE', False, [512, 512], 360, False],
-        ['CORD', True, [512, 512], 800, True],
-        ['CORD', False, [512, 512], 100, False],
-        ['DocArtefacts', True, [512, 512], 2700, False],
-        ['DocArtefacts', False, [512, 512], 300, True],
-        ['IIIT5K', True, [32, 128], 2000, True],
-        ['IIIT5K', False, [32, 128], 3000, False],
-        ['SVT', True, [512, 512], 100, True],
-        ['SVT', False, [512, 512], 249, False],
+        ['FUNSD', True, [512, 512], 149, False, True],
+        ['FUNSD', False, [512, 512], 50, True, True],
+        ['SROIE', True, [512, 512], 626, False, True],
+        ['SROIE', False, [512, 512], 360, False, True],
+        ['CORD', True, [512, 512], 800, True, True],
+        ['CORD', False, [512, 512], 100, False, True],
+        ['DocArtefacts', True, [512, 512], 2700, False, True],
+        ['DocArtefacts', False, [512, 512], 300, True, True],
+        ['IIIT5K', True, [32, 128], 2000, True, True],
+        ['IIIT5K', False, [32, 128], 3000, False, True],
+        ['SVT', True, [512, 512], 100, True, True],
+        ['SVT', False, [512, 512], 249, False, True],
+        ['SynthText', True, [512, 512], 27, False, False],
+        ['SynthText', False, [512, 512], 3, True, False],
     ],
 )
-def test_dataset(dataset_name, train, input_size, size, rotate):
+def test_dataset(dataset_name, train, input_size, size, rotate, download):
 
     ds = datasets.__dict__[dataset_name](
-        train=train, download=True, sample_transforms=Resize(input_size), rotated_bbox=rotate,
+        train=train, download=download, sample_transforms=Resize(input_size), rotated_bbox=rotate,
```

Comment:
better to do an exception to create the `ds` object like
```python
if dataset_name.lower() == "synthtext":
...
```

The idea here is the following:
- if it's synthtext, we set `download=False`
- before calling the constructor, we download the subsample, put it in the cache
- when we call the constructor, we need to monkeypatch the expected hash

Reply:
@fg-mindee 
Where to download the subsample (URL) ?? :sweat_smile: 
Can you take a look at the monkeypatch i have never used / needed this before :smiley: